lib/libc/minimal/source/stdlib/exit.c

```diff
@@ -10,6 +10,6 @@
 void _exit(int status)
 {
 	printk("exit\n");
-	while (1) {
+	for (;;) {
```

Comment:
Just curious, why not just `while (true)`? Do we have a rule prohibiting the use of `while (true)`?

Reply:
I was actually wondering the same thing. I haven't been able to locate any resources that says that MISRA Rule 14:4 requires this.

Reply:
while (true) would be compliant as well, but for (;;) is usually preferred to express infinite (unless the body make them finite) loop.

Reply:
Not in Linux (and Zephyr is trying to stay close to that coding style). I'd suggest using `while (true)` here instead, since that's the most common pattern today.

Reply:
Please try

$ git grep "for (;;)"

There are 41 matches now: changing them all to "while (true)" is certainly doable, but it has nothing to do with compliance to the MISRA rule that is relevant in this discussion.

Reply:
The two instances of `while (1)' that I previously changed to `for (;;)' are `while (true)' now.

Reply:
> Please try
> 
> $ git grep "for (;;)"
> 
> There are 41 matches now: changing them all to "while (true)" is certainly doable, but it has nothing to do with compliance to the MISRA rule that is relevant in this discussion.

Tried it!

```
olof@rip:~/src/z/zephyr (v2.7-auditable-branch) $ git grep "while (true)"|wc -l
173
olof@rip:~/src/z/zephyr (v2.7-auditable-branch) $ git grep "while (1)"|wc -l
374
olof@rip:~/src/z/zephyr (v2.7-auditable-branch) $ git grep "for (;;)"|wc -l
41
```

So, changing it to the common convention, even if there are a few of the other kind, seems reasonable.

It's not about whether it fixes the MISRA warning or not, it's whether there'll be another pass over all of this because you didn't approach it with merge-friendly style for when it's heading into the main branch.

Doing the work twice seems counterproductive, which is why we're all looking at some of these first few patches and comment closer on them. I'd expect feedback to slow down once the approach/expectation gets more aligned.

Reply:
(And nobody is asking for existing `for (;;)` to be changed, just not to add any new ones.